lib/ts/recipe/oauth2provider/recipeImplementation.ts

```diff
@@ -312,19 +315,20 @@ export default function getRecipeInterface(
             }
```

Comment:
we should remove the above workaround and pass the token payloads as params to the core call instead of calling `saveTokensForHook` (which should be removed)